### ISSUE TITLE
Support for Razer [Razer Mouse Dock Pro +Razer Basilisk V3 Pro] 

### ISSUE THREAD

**Issue by DarkyAndSparky** [openrazer/razer-drivers#2350]
### Device Name

Razer Mouse Dock Pro

### Product Number

RC21-01990100-0000

### Model Number

PM2438U28101525

### VID:PID

1532:00A4

### USB Details

```shell
Bus 003 Device 002: ID 1532:00a4 Razer USA, Ltd Razer Mouse Dock Pro
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 [unknown]
  bDeviceSubClass         0 [unknown]
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x00a4 Razer Mouse Dock Pro
  bcdDevice            1.01
  iManufacturer           1 Razer
  iProduct                2 Razer Mouse Dock Pro
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xe0
      Self Powered
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
          Report Descriptors: 
            ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 [unknown]
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     314
          Report Descriptors: 
            ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0010  1x 16 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
          Report Descriptors: 
            ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
```

### Device Input Paths

```shell
итого 0
drwxr-xr-x 2 root root 180 ноя 28 16:05 .
drwxr-xr-x 4 root root 620 ноя 28 16:05 ..
lrwxrwxrwx 1 root root   9 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-event-if01 -> ../event9
lrwxrwxrwx 1 root root   9 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-event-mouse -> ../event6
lrwxrwxrwx 1 root root   9 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-if01-event-kbd -> ../event7
lrwxrwxrwx 1 root root   9 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-if01-event-mouse -> ../event8
lrwxrwxrwx 1 root root   9 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-if01-mouse -> ../mouse1
lrwxrwxrwx 1 root root  10 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-if02-event-kbd -> ../event10
lrwxrwxrwx 1 root root   9 ноя 28 16:05 usb-Razer_Razer_Mouse_Dock_Pro-mouse -> ../mouse0
```

### USB HID Dump

```shell
libusb: error [get_usbfs_fd] libusb couldn't open USB device /dev/bus/usb/003/002, errno=13
libusb: error [get_usbfs_fd] libusb requires write access to USB device nodes
Failed to find and open the devices: Access denied (insufficient permissions)
```

### Packet Captures

![image](https://github.com/user-attachments/assets/dbf23a3f-044c-4135-af83-48efd0abeb84)

**Comment by DarkyAndSparky** [comment]
DEAMON 3.9.0
PYTHON LIBRAARY 3.9.0
D-BUS 1.14.10
DKMS 3.0.12

System:
  Kernel: 6.10.13-3-MANJARO arch: x86_64 bits: 64 compiler: gcc v: 14.2.1
  Desktop: KDE Plasma v: 6.1.5 Distro: Manjaro base: Arch Linux
Machine:
  Type: Laptop System: ASUSTeK product: ASUS TUF Gaming A15 FA507NV_FA507NV
    v: 1.0 serial: <superuser required>
  Mobo: ASUSTeK model: FA507NV v: 1.0 serial: <superuser required>
    UEFI: American Megatrends LLC. v: FA507NV.315 date: 10/18/2024
Battery:
  ID-1: BAT1 charge: 82.3 Wh (100.0%) condition: 82.3/90.2 Wh (91.2%)
    volts: 17.1 min: 15.9 model: ASUS A32-K55 status: full
CPU:
  Info: 8-core model: AMD Ryzen 7 7735HS with Radeon Graphics bits: 64
    type: MT MCP arch: Zen 3+ rev: 1 cache: L1: 512 KiB L2: 4 MiB L3: 16 MiB
  Speed (MHz): avg: 400 min/max: 400/4829 cores: 1: 400 2: 400 3: 400 4: 400
    5: 400 6: 400 7: 400 8: 400 9: 400 10: 400 11: 400 12: 400 13: 400 14: 400
    15: 400 16: 400 bogomips: 102255
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: NVIDIA AD107M [GeForce RTX 4060 Max-Q / Mobile] vendor: ASUSTeK
    driver: nvidia v: 550.120 arch: Lovelace bus-ID: 01:00.0
  Device-2: Advanced Micro Devices [AMD/ATI] Rembrandt [Radeon 680M]
    vendor: ASUSTeK driver: amdgpu v: kernel arch: RDNA-2 bus-ID: 36:00.0
    temp: 43.0 C
  Device-3: Sonix USB2.0 HD UVC WebCam driver: uvcvideo type: USB
    bus-ID: 1-4:2
  Display: x11 server: X.Org v: 21.1.14 with: Xwayland v: 24.1.4 driver: X:
    loaded: amdgpu,nvidia unloaded: modesetting,nouveau dri: radeonsi
    gpu: amdgpu resolution: 1920x1080~144Hz
  API: EGL v: 1.5 drivers: kms_swrast,nvidia,radeonsi,swrast platforms:
    active: gbm,x11,surfaceless,device inactive: wayland,device-2
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: amd mesa v: 24.2.4-arch1.0.1
    glx-v: 1.4 direct-render: yes renderer: AMD Radeon 680M (radeonsi rembrandt
    LLVM 18.1.8 DRM 3.57 6.10.13-3-MANJARO)
  API: Vulkan v: 1.3.295 drivers: nvidia,radv surfaces: xcb,xlib devices: 2
Audio:
  Device-1: NVIDIA vendor: ASUSTeK driver: snd_hda_intel v: kernel
    bus-ID: 01:00.1
  Device-2: Advanced Micro Devices [AMD/ATI] Rembrandt Radeon High
    Definition Audio driver: snd_hda_intel v: kernel bus-ID: 36:00.1
  Device-3: Advanced Micro Devices [AMD] ACP/ACP3X/ACP6x Audio Coprocessor
    vendor: ASUSTeK driver: snd_pci_acp6x v: kernel bus-ID: 36:00.5
  Device-4: Advanced Micro Devices [AMD] Family 17h/19h HD Audio
    vendor: ASUSTeK driver: snd_hda_intel v: kernel bus-ID: 36:00.6
  API: ALSA v: k6.10.13-3-MANJARO status: kernel-api
  Server-1: JACK v: 1.9.22 status: off
  Server-2: PipeWire v: 1.2.5 status: active
Network:
  Device-1: Realtek RTL8111/8168/8211/8411 PCI Express Gigabit Ethernet
    vendor: ASUSTeK driver: r8169 v: kernel port: e000 bus-ID: 03:00.0
  IF: eno1 state: up speed: 100 Mbps duplex: full mac: <filter>
  Device-2: Realtek RTL8852BE PCIe 802.11ax Wireless Network
    vendor: AzureWave driver: rtw89_8852be v: kernel port: d000 bus-ID: 04:00.0
  IF: wlp4s0 state: down mac: <filter>
Bluetooth:
  Device-1: IMC Networks Bluetooth Radio driver: btusb v: 0.8 type: USB
    bus-ID: 3-3:3
  Report: rfkill ID: hci0 rfk-id: 0 state: up address: see --recommends
Drives:
  Local Storage: total: 2.29 TiB used: 19.14 GiB (0.8%)
  ID-1: /dev/nvme0n1 vendor: Crucial model: CT2000P310SSD2 size: 1.82 TiB
    temp: 36.9 C
  ID-2: /dev/nvme1n1 vendor: Micron model: 2400 MTFDKBA512QFM
    size: 476.94 GiB temp: 30.9 C
Partition:
  ID-1: / size: 1.79 TiB used: 19.13 GiB (1.0%) fs: ext4 dev: /dev/nvme0n1p2
  ID-2: /boot/efi size: 299.4 MiB used: 292 KiB (0.1%) fs: vfat
    dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: file size: 512 MiB used: 0 KiB (0.0%) file: /swapfile
Sensors:
  System Temperatures: cpu: 47.5 C mobo: N/A gpu: amdgpu temp: 43.0 C
  Fan Speeds (rpm): cpu: 0
Info:
  Memory: total: 32 GiB note: est. available: 30.6 GiB used: 3.47 GiB (11.4%)
  Processes: 353 Uptime: 19m Init: systemd
  Packages: 1284 Compilers: gcc: 14.2.1 Shell: Zsh v: 5.9 inxi: 3.3.36

**Comment by z3ntu** [comment]
Duplicate.

For Mouse Dock Pro: #2060

For Basilisk V3 Pro check the search, your device doesn't show up in your lsusb output.
